From e8db929ae763e7dce5bafa6abc5ed8fb2ebe0189 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Tue, 17 Apr 2018 08:43:02 -0500
Subject: [PATCH] "fix-symlinked-device-tree _______________________________

*Previous From a325f7e535e355044c2480a2f4f6df8d47f8d404 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Wed, 28 Feb 2018 08:39:56 -0600
*Previous Subject: [PATCH] "fix-symlinked-device-tree _______________________________
*
**Previous From c2def5883bb7d3b64ba516c19580ac88548cf09e Mon Sep 17 00:00:00 2001
**Previous From: Meticulus <theonejohnnyd@gmail.com>
**Previous Date: Wed, 22 Feb 2017 10:39:56 -0600
**Previous Subject: [PATCH] fix-symlinked-device-tree
**""
---
 core/envsetup.mk       | 4 ++--
 core/product_config.mk | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/envsetup.mk b/core/envsetup.mk
index d9842e3..0c83a0e 100644
--- a/core/envsetup.mk
+++ b/core/envsetup.mk
@@ -216,8 +216,8 @@ SDK_HOST_ARCH := x86
 board_config_mk := \
 	$(strip $(sort $(wildcard \
 		$(SRC_TARGET_DIR)/board/$(TARGET_DEVICE)/BoardConfig.mk \
-		$(shell test -d device && find -L device -maxdepth 4 -path '*/$(TARGET_DEVICE)/BoardConfig.mk') \
-		$(shell test -d vendor && find -L vendor -maxdepth 4 -path '*/$(TARGET_DEVICE)/BoardConfig.mk') \
+		$(shell test -d device && find -L device -follow -maxdepth 4 -path '*/$(TARGET_DEVICE)/BoardConfig.mk') \
+		$(shell test -d vendor && find -L vendor -follow -maxdepth 4 -path '*/$(TARGET_DEVICE)/BoardConfig.mk') \
 	)))
 ifeq ($(board_config_mk),)
   $(error No config file found for TARGET_DEVICE $(TARGET_DEVICE))
diff --git a/core/product_config.mk b/core/product_config.mk
index aec296c..d57a3a8 100644
--- a/core/product_config.mk
+++ b/core/product_config.mk
@@ -170,7 +170,7 @@ include $(BUILD_SYSTEM)/device.mk
 
 # An RR build needs only the RR product makefiles.
 ifneq ($(RR_BUILD),)
-  all_product_configs := $(shell find device -path "*/$(RR_BUILD)/rr.mk")
+  all_product_configs := $(shell find -L device -path "*/$(RR_BUILD)/rr.mk")
   all_product_configs += $(wildcard vendor/rr/build/target/product/rr_$(RR_BUILD).mk)
 else
   ifneq ($(strip $(TARGET_BUILD_APPS)),)
-- 
2.7.4

